### ISSUE TITLE
failing parallel tests always succeed

### ISSUE THREAD

**Issue by fommil** [sbt/sbt#1890]
This is the dual of the bug reported at scalatest https://github.com/scalatest/scalatest/issues/432 which has a reproducible project.

I'm seeing this in 0.13.7 in ensime-server and it is really quite a serious bug that tests are being reported as successful when they have actually failed!

**Comment by fommil** [comment]
(sorry, that wasn't a workaround... referenced the wrong ticket)

Note that this happens with or without the parallel forking.

**Comment by eed3si9n** [comment]
Is this a sbt issue or scalatest issue?

**Comment by fommil** [comment]
I have no idea, nor any way to tell.

**Comment by fommil** [comment]
I'm still seeing this in 0.13.8

```
[info] Run completed in 4 minutes, 56 seconds.
[info] Total number of tests run: 69
[info] Suites: completed 9, aborted 0
[info] Tests: succeeded 66, failed 3, canceled 0, ignored 1, pending 0
[info] *** 3 TESTS FAILED ***
[info] Passed: Total 22, Failed 0, Errors 0, Passed 22
```

even with 

```
parallelExecution in IntegrationTest := false,
testForkedParallel in IntegrationTest := false,
```

there is definitely something very wrong, and I don't know if it's sbt or scalatest @bvenners

**Comment by fommil** [comment]
@eed3si9n this is marked "needs info" but I have given you a test case. Just do an `assert(false)` in any of the ensime-server integration tests and watch the suite pass...

**Comment by eed3si9n** [comment]
By NeedInfo, I meant needs-further-research-into-what's-going-on.

**Comment by fommil** [comment]
aah ok. It's extremely concerning actually.

**Comment by fommil** [comment]
@eed3si9n see the corresponding scalatest ticket. Information posted there suggests that this is an sbt bug rather than a scalatest runner bug.

**Comment by fommil** [comment]
I just seen this even with

``` scala
parallelExecution in IntegrationTest := false,
testForkedParallel in IntegrationTest := false,
```

this is starting to become a really serious defect in sbt. There is a lot more chatter going on in the sister bug report at https://github.com/scalatest/scalatest/issues/432 where some reporters have investigated and shown that it is an sbt, not a scalatest, bug.

**Comment by fommil** [comment]
`with ParallelTestExecution` may be the culprit! The tests seems to stabilise when I remove it. So maybe `parallelExecution in IntegrationTest` is ignored.

**Comment by fommil** [comment]
@eed3si9n responding to a query you had on gitter after I'd left, for me this is with `fork=true`. (Although I'm having a hell of a time trying to fork more than one JVM as a way of getting some parallelism in my tests)

**Comment by fommil** [comment]
I'm happy to report that this is a bug in scalatest and it will be fixed in 2.2.6!
